test/timetable.test.js

```diff
@@ -0,0 +1,197 @@
+  expected.offset = 0
+
+  t.deepEqual(tt.createTimeline(grid), expected, 'creates timeline correctly with faculty grid')
+
+  t.end()
+})
+
+test('calculateEventPosition()', t => {
+  const timeline = {
+    start: 27000,
+    end: 76500,
+    duration: 49500,
+    hourDuration: 3600,
+    hours: 13.75,
+    firstHour: 8,
+    offset: 0,
+  }
+
+  const events = [
+    { startsAt: '1970-01-01 07:30:00', endsAt: '1970-01-01 21:15:00', expected: { position: 0, length: 1 } },
+    { startsAt: '1970-01-01 14:22:30', endsAt: '1970-01-01 21:15:00', expected: { position: 0.5, length: 0.5 } },
+    { startsAt: '1970-01-01 14:22:30', endsAt: '1970-01-01 17:48:45', expected: { position: 0.5, length: 0.25 } },
+  ]
+
+  R.forEach(event => {
+    t.deepEqual(tt.calculateEventPosition(event, timeline), event.expected, 'calculates correct event position')
+  }, events)
+
+  t.end()
+})
+
+test('calculateHourLabels()', t => {
+  const timeline = {
+    start: 27000,
+    end: 77400,
+    duration: 50400,
+    hourDuration: 3600,
+    hours: 14,
+    firstHour: 8,
+    offset: 0.5,
+  }
+
+  // Test n, n+1, m-1, m (n=first, m=last)
+  const expected = [
+    {
+      id: 'hlbl-0',
+      label: 8,
+      position: 0.03571428571428571,
+      length: 0.07142857142857142,
+    },
+    {
+      id: 'hlbl-1',
+      label: 9,
+      position: 0.10714285714285714,
+      length: 0.07142857142857142,
+    },
+    {
+      id: 'hlbl-12',
+      label: 20,
+      position: 0.8928571428571429,
+      length: 0.07142857142857142,
+    },
+    {
+      id: 'hlbl-13',
+      label: 21,
+      position: 0.9642857142857143,
+      length: 0.07142857142857142,
+    },
+  ]
+
+  const result = tt.calculateHourLabels(timeline)
+  const rlen = result.length
+  const resultSelection = [result[0], result[1], result[rlen - 2], result[rlen - 1]]
+
+  t.deepEqual(resultSelection, expected, 'calculate hour labels for timeline 7:30 - 21:15')
+
+  t.end()
+})
+
+test('mapPropertiesToClass()', t => {
+  const properties = {
+    isA: true,
+    isB: false,
+    isProperty: true,
+    isLoremIpsum: false,
+    isDolorSitAmet: true,
+  }
+
+  const expected = 'class class--a class--property class--dolorsitamet '
+
+  t.equal(tt.mapPropertiesToClass(properties, 'class'), expected, 'returns properties transformed to classname')
+  t.end()
+})
+
+test('groupEventsByDays()', t => {
+  const events = [
+    {startsAt: '2015-12-03 12:00:00'},
+    {startsAt: '2015-12-03 14:00:00'},
+    {startsAt: '2015-12-04 12:00:00'},
+    {startsAt: '2015-12-05 12:00:00'},
+  ]
+
+  const expected = {
+    3: [
+      {startsAt: '2015-12-03 12:00:00'},
+      {startsAt: '2015-12-03 14:00:00'},
+    ],
+    4: [
+      {startsAt: '2015-12-04 12:00:00'},
+    ],
+    5: [
+      {startsAt: '2015-12-05 12:00:00'},
+    ],
+  }
+
+  t.deepEqual(tt.groupEventsByDays(events), expected, 'groups events by weekdaynum')
+
+  t.end()
+})
+
+test('calculateOverlap()', t => {
+  const expected = [
```

Comment:
And again… https://github.com/cvut/fittable/commit/64ae07c6a88b001e655000498f8b64d492d18914#commitcomment-14958484. Don’t you see this horrible code duplication?